docs/development/module/off-chain-workers.md

```diff
@@ -0,0 +1,383 @@
+    call: Call,
+    public: Self::Public,
+    account: AccountId,
+    index: Index,
+  ) -> Option<(Call, <UncheckedExtrinsic as sp_runtime::traits::Extrinsic>::SignaturePayload)> {
+    let period = 1 << 8;
+    let current_block = System::block_number().saturated_into::<u64>();
+    let tip = 0;
+    let extra: SignedExtra = (
+      system::CheckVersion::<Runtime>::new(),
+      system::CheckGenesis::<Runtime>::new(),
+      system::CheckEra::<Runtime>::from(generic::Era::mortal(period, current_block)),
+      system::CheckNonce::<Runtime>::from(index),
+      system::CheckWeight::<Runtime>::new(),
+      transaction_payment::ChargeTransactionPayment::<Runtime>::from(tip),
+    );
+    let raw_payload = SignedPayload::new(call, extra).ok()?;
+    let signature = TSigner::sign(public, &raw_payload)?;
+    let address = Indices::unlookup(account);
+    let (call, extra, _) = raw_payload.deconstruct();
+    Some((call, (address, signature, extra)))
+  }
+}
+```
+
+Inside the `contruct_runtime!` macro where you put all various packages as your runtime enum, add an additional parameter `ValidateUnsigned` if you are using unsigned transactions in off-chain workers.
+
+**runtime/src/lib.rs**
+
+```rust
+construct_runtime!(
+  pub enum Runtime where
+    Block = Block,
+    NodeBlock = opaque::Block,
+    UncheckedExtrinsic = UncheckedExtrinsic
+  {
+    // ...
+    // ...
+    // For using unsigned transactions
+    MyRuntime: my_runtime::{ Module, Call, Storage, Event<T>, transaction_validity::ValidateUnsigned }
+
+    // For just using signed transactions, it can just well be:
+    // MyRuntime: my_runtime::{ Module, Call, Storage, Event<T> }
+  }
+);
+```
+
+## Insert Keys in `service.rs`
+
+We have specify a local keystore with `KeyTypeId` to store app-specific sub-keys that are accessible by off-chain workers for signing transactions. But currently there is no sub-keys inside. We can add it in one of the following two ways.
+
+### Development: Add the First User Key as App Subkey
+
+In development environment, you can add the first user key as the app sub-key. Update the `service.rs` as followed.
+
+**node/src/service.rs**
+
+```rust
+pub fn new_full<C: Send + Default + 'static>(config: Configuration<C, GenesisConfig>)
+  -> Result<impl AbstractService, ServiceError>
+{
+  // ...
+  // ...
+  // Add this line
+  let dev_seed = config.dev_key_seed.clone();
+
+  // ...
+  // ...
+  let service = builder.with_network_protocol(|_| Ok(NodeProtocol::new()))?
+    .with_finality_proof_provider(|client, backend|
+      Ok(Arc::new(GrandpaFinalityProofProvider::new(backend, client)) as _)
+    )?
+    .build()?;
+
+  // Add the following section
+  if let Some(seed) = dev_seed {
+    service
+      .keystore()
+      .write()
+      .insert_ephemeral_from_seed_by_type::<runtime::my_runtime::crypto::Pair>(
+        &seed,
+        runtime::my_runtime::KEY_TYPE,
+      )
+      .expect("Dev Seed should always succeed.");
+  }
+}
```

Comment:
What key or keys actually get added here? Just Alice? Alice, Bob, etc? Something else?

Reply:
Referring to: https://github.com/paritytech/substrate/blob/master/client/cli/src/lib.rs#L901-L908

In development, it is only Alice.